File: Searching Algorithms/Linear Search.cpp
#include <iostream>

using namespace std;

int linearSearch( int A[], int N,int n );

int main()
{
    int N;

    cout << "Enter No of Elements in array : ";
    cin >> N;

    int arr[N];
    int n;

    cout << "Enter elements of array : ";
    for ( int i = 0; i < N; i++ )
    {
        cin >> arr[i];
    }

    cout << "Enter the number to be searched : ";
    cin >> n;

    int pos = linearSearch( arr, N ,n );

    if( pos > 0 )
    {
        cout << "Elements is found at position " << pos << endl;
    }else
    {
        cout << "Element not found" << endl;
    }

    return 0;
}

int linearSearch( int A[], int N, int n )
{
    int loc = -1;

    for ( int i = 0; i < N; i++ )
    {
        if ( A[i] == n )
        {
            loc = i;
            break;
        }
    }

    return loc;
}

File: Searching Algorithms/Python/Linear Search.py
def linearSearch( arr, len , num):
    for i in range( len ) :
        if arr[i] == num :
            return i
    else:
        return -1

len = int( input( "Enter Length of Array : " ) )

arr = []

for i in range( len ) :
    arr.append( int( input( "Enter Value in Array : " ) ) )

num = int( input( "Enter Number to find in array : " ) )

loc = linearSearch( arr, len , num )

if loc == -1 :
    print( "Number is Not found in Given array" )
else:
    print( "Number is found at index :", loc )